src/Gaffer/Spreadsheet.cpp

```diff
@@ -0,0 +1,532 @@
+
+GAFFER_PLUG_DEFINE_TYPE( Spreadsheet::RowsPlug );
+
+Spreadsheet::RowsPlug::RowsPlug( const std::string &name, Direction direction, unsigned flags )
+	:	ValuePlug( name, direction, flags )
+{
+	RowPlugPtr defaultRow = new RowPlug( "default" );
+	addChild( defaultRow );
+}
+
+size_t Spreadsheet::RowsPlug::addColumn( const ValuePlug *value, IECore::InternedString name )
+{
+	if( name.string().empty() )
+	{
+		name = value->getName();
+	}
+
+	for( auto &row : RowPlug::Range( *this ) )
+	{
+		CellPlugPtr cellPlug = new CellPlug( name, value );
+		cellPlug->valuePlug()->setFrom( value );
+		row->cellsPlug()->addChild( cellPlug );
+	}
+	if( auto *o = outPlug() )
+	{
+		PlugPtr outColumn = value->createCounterpart( name, Plug::Out );
+		outColumn->setFlags( Plug::Dynamic, false );
+		o->addChild( outColumn );
+	}
+	return getChild<RowPlug>( 0 )->cellsPlug()->children().size() - 1;
+
+}
+
+void Spreadsheet::RowsPlug::removeColumn( size_t columnIndex )
+{
+	if( columnIndex >= getChild<RowPlug>( 0 )->cellsPlug()->children().size() )
+	{
+		throw IECore::Exception( "Column index out of range" );
+	}
+
+	for( auto &row : RowPlug::Range( *this ) )
+	{
+		row->cellsPlug()->removeChild( row->cellsPlug()->getChild( columnIndex ) );
+	}
+	if( auto *o = outPlug() )
+	{
+		o->removeChild( o->getChild( columnIndex ) );
+	}
+}
+
+Spreadsheet::RowPlug *Spreadsheet::RowsPlug::addRow()
+{
+	RowPlugPtr newRow = new RowPlug( "row1" );
+	for( auto &defaultCell : CellPlug::Range( *getChild<RowPlug>( 0 )->cellsPlug() ) )
+	{
+		CellPlugPtr newCell = boost::static_pointer_cast<CellPlug>(
+			defaultCell->createCounterpart( defaultCell->getName(), Plug::In )
+		);
+		newCell->setFrom( defaultCell.get() );
+		newRow->cellsPlug()->addChild( newCell );
+	}
+	addChild( newRow );
+	return newRow.get();
+}
+
+void Spreadsheet::RowsPlug::addRows( size_t numRows )
+{
+	for( ; numRows; --numRows )
+	{
+		addRow();
+	}
+}
+
+Gaffer::ValuePlug *Spreadsheet::RowsPlug::outPlug()
+{
+	// Find the `outPlug()` on our parent Spreadsheet node.
+	// We use this from `add/removeColumn()` so that we can
+	// add the appropriate columns to the output plug. It's
+	// not ideal that this responsibility falls to us. What
+	// might be nice would be to rename RowsPlug to SheetPlug
+	// and let it manage both "rows" and "out" children, so
+	// that it's more natural to modify "out". We can't do that
+	// at present because we don't allow connections to plugs
+	// which have a mix of input and output children.
+	if( auto *s = parent<Spreadsheet>() )
+	{
+		return s->outPlug();
+	}
+	return nullptr;
+}
+
+
+Gaffer::PlugPtr Spreadsheet::RowsPlug::createCounterpart( const std::string &name, Direction direction ) const
+{
+	RowsPlugPtr result = new RowsPlug( name, direction, getFlags() );
+	for( const auto &row : RowPlug::Range( *this ) )
+	{
+		result->setChild( row->getName(), row->createCounterpart( row->getName(), direction ) );
```

Comment:
Added a comment for the next version :

```
// Using `setChild()` rather than `addChild()` because we want to replace
// the default-constructed default row with one with the right number of
// cells.
```